### PR TITLE
SITL mission tests (via MAVROS)

### PR DESCRIPTION
**Pull request by AndreasAntener**
@dagar This adds SITL mission testing for CI, via MAVROS

Improvements we should make:
- It would be good to move this to plain MAVLink
- Switch to new JSON mission format

### CODE REVIEW

**Comment by AndreasAntener** [comment]
Will re-open again when segfault is gone

**Comment by AndreasAntener** [comment]
Re-opened without running on CI for the moment.

**Comment by AndreasAntener** [comment]
@dagar mission testing works on Jenkins: http://sitl01.dronetest.io/job/PX4%20Firmware%20POSIX%20SITL%20PRs/1188/ :)

One test failed but you don't see for which mission so I'm going to try and make that visible. After we should have a good starting point to iterate on.

**Comment by LorenzMeier** [comment]
Awesome work! Let's get this in!

**Comment by LorenzMeier** [comment]
Rebased and applied.

File: integrationtests/python_src/px4_it/mavros/mission_test.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        for waypoint in wps:
+            # only check position for waypoints where this makes sense
+            if waypoint.frame == Waypoint.FRAME_GLOBAL_REL_ALT or waypoint.frame == Waypoint.FRAME_GLOBAL:
+                alt = waypoint.z_alt
+                if waypoint.frame == Waypoint.FRAME_GLOBAL_REL_ALT:
+                    alt += self.home_alt
+                self.reach_position(waypoint.x_lat, waypoint.y_long, alt, 600, index)
+
+            # check if VTOL transition happens if applicable
+            if waypoint.command == 84 or waypoint.command == 85 or waypoint.command == 3000:
```
Comment (vooon) [inline integrationtests/python_src/px4_it/mavros/mission_test.py]: `waypoint.command in (84, 85, 3000)` a bit better, do you think?

**Pull request closed without merging**